src/bzflag/ServerLink.cxx

```diff
@@ -491,8 +482,69 @@ void            ServerLink::send(uint16_t code, uint16_t len,
 #endif
 #endif //WIN32
 
+int ServerLink::fillTcpReadBuffer()
+{
+    int emptySpace = sizeof(tbuf) - tcpBufferPos;
+    if (!emptySpace)
+        return 0;
+    ssize_t rlen = recv(fd, &tbuf[tcpBufferPos], emptySpace, 0);
+    if (rlen < 0)
+    {
+        if (errno == EAGAIN)
+            return 0;
+        else if (errno == EWOULDBLOCK)
+            return 0;
+        else
+            return -1;
+    }
+
+    if (rlen)
+    {
+#if defined(NETWORK_STATS)
+        bytesReceived += rlen;
+        packetsReceived++;
+#endif
+        tcpBufferPos += rlen;
+    }
+    return 0;
+}
+
+void ServerLink::consumeTcpReadBuffer(unsigned int len)
+{
+    int remain  = tcpBufferPos - len;
+
+    if (remain)
+        memmove(tbuf, &tbuf[len], remain);
+    tcpBufferPos -= len;
+}
+
+// A packet is there?
+bool ServerLink::tcpPacketIn()
+{
+    // Enough data to decode the header?
+    if (tcpBufferPos < 4)
+        return false;
+
+    // unpack len
+    const void* buf = tbuf;
+    uint16_t    tmpLen;
+    buf = nboUnpackUShort(buf, tmpLen);
+
+    // Add header size
+    tmpLen += 4;
+
+    // Check if len is good
+    if (tmpLen > MaxPacketLen)
+        // Len is bigger than the Max Packet Len
+        // The packet is there but it is wrong
+        return true;
+
+    // Check if we have enough data for the whole message
+    return tcpBufferPos >= tmpLen;
+}
+
 int         ServerLink::read(uint16_t& code, uint16_t& len,
-                             void* msg, int blockTime)
+                             void* msg, int UNUSED(blockTime))
```

Comment:
since you are changing the networking layer, I'd like to see you make a breaking change. Don't flag this argument as unused; remove it.